### ISSUE TITLE
Request support for pre-processing "full" when using JDK >= 23

### ISSUE THREAD

**Issue by jmborer** [apache/netbeans-mavenutils-nbm-maven-plugin#201]
As the pre-processing has changed since JDK 23 (https://inside.java/2024/06/18/quality-heads-up/) and is required by many NB platform functions, the NB Maven plugin shall force the compiler settings to:
```
<configuration>
      <proc>full</proc>
</configuration>
```
Or at least inform the user that it shall turn it to "full".

**Comment by neilcsmith-net** [comment]
We should probably see if there's a way to provide a better warning.  I don't think we should be changing the configuration, even if that's possible.  @ebarboni what do you think on this one?

**Comment by ebarboni** [comment]
I don't think nbm plugin can alter configuration for compiler plugin. Maybe a hint in the IDE could help as well of having a condition in archetypes to add this line according to JDK version.

**Comment by neilcsmith-net** [comment]
Thanks!  That's what I thought.  Hint might be useful for pre-existing projects.  Should move that request over to the IDE issue queue.

No need to add the archetype change according to version - should be OK to add in on all supported JDKs.  There's a bunch of things that need updating in the archetypes already.

**Comment by jmborer** [comment]
Indeed, this is what I would expect as well. Conditional Maven profiles could be used as well to turn on the proper flag for the compiler. This could be generated by the archetype is a non-invasive way to provide proper flags to build NB based applications.

PS: I turned the proc to "full" on my projects with JDK23 and it works as before.

**Comment by neilcsmith-net** [comment]
@jmborer feel free to submit a PR for either.  No need for conditional flags - this will work fine on all supported JDKs.
